app/components/UserBets.tsx

```diff
@@ -179,9 +180,14 @@ export const UserBets = ({ market }: UserBets) => {
 
           const alreadyClaimed = canClaim && outcomesBalance[index] === BigInt(0);
 
-          const collateralSpent = outcomesCollateralAmountUSDSpent[index].toFixed(2);
-          const tradedBalance = outcomesTradedBalance[index].toFixed(2);
-
+          const collateralSpent = formatEtherWithFixedDecimals(
+            parseEther(outcomesCollateralAmountUSDSpent[index].toFixed(20)),
+            4
+          );
+          const tradedBalance = formatEtherWithFixedDecimals(
+            parseEther(outcomesTradedBalance[index].toFixed(20)),
+            4
+          );
```

Comment:
I think here we are complicating just to feat the function parameters. What about having two functions? One for normal numbers and other for wei?

Reply:
We are making the exception as the rule. I would create two functions, one that fixes decimals, and on that formatsEther and call the on that fixes decimals.

Reply:
that makes sense, I'll explore that.